lib/matplotlib/patches.py

```diff
@@ -1940,18 +1940,45 @@ class _Base:
         ``__init__`` method because they must be able to be initialized
         without arguments.
 
-        Subclasses must implement the `transmute` method. It receives the
+        Subclasses must implement the `__call__` method. It receives the
         enclosing rectangle *x0, y0, width, height* as well as the
         *mutation_size*, which scales the outline properties such as padding.
         It returns the outline of the fancy box as `.path.Path`.
         """
 
+        @cbook.deprecated("3.4")
         def transmute(self, x0, y0, width, height, mutation_size):
             """Return the `~.path.Path` outlining the given rectangle."""
-            raise NotImplementedError('Derived must override')
-
-        def __call__(self, x0, y0, width, height, mutation_size,
-                     aspect_ratio=1.):
+            return self(self, x0, y0, width, height, mutation_size, 1)
```

Comment:
yes, but subclasses previously needed to override `__call__` to get any meaningful behavior, no?  previously not overriding it would result in a NotImplementedError from the base class, now you'd get a TypeError at argument binding time but I think that's close enough?

Reply:
I thought they had to override `transmute` before?

Reply:
Sorry, I got it wrong just above.  Users had to override *transmute* before, which means that the implementation of _Base.transmute is in facto mostly irrelevant for backcompat purposes.  I guess I could even leave it as raising a NotImplementedError? (although I would change the error message, which would be confusing)

Reply:
OK, but now nothing calls `transmute`? So how would existing subclassers find out the deprecation?

Reply:
hum... I guess that in `__call__` I can add a call to `_deprecate_method_override(__class__.transmute, ...)` (and put back the old version of the code under the `if transmute := _deprecate_method_override(...)` branch)?

Reply:
Yea, if that works, sounds good.

Reply:
New version in, but the whole thing's a bit tricky...